### PR TITLE
Added check for multi-result pages to prevent null exception error.

### PR DESCRIPTION
**Pull request by CBXZero**

### CODE REVIEW

**Comment by CBXZero** [comment]
First time I attempted the squash, I followed a stackoverflow answer that was misleading. This should be good to go and is all bundled in 1 commit :panda_face:

**Comment by davidchambers** [comment]
Thank you for persevering despite your version control troubles. :)

Please `git commit --amend` to give the commit a sensible message. You'll then need to run `git push origin master --force` to update this pull request.

**Comment by CBXZero** [comment]
Updated the commit, didn't update the pull request title though :/

File: src/gatherer.coffee [hunk 1]
```diff
@@ -32,7 +32,7 @@ gatherer.request = (args...) ->
       callback err
     else if res.statusCode isnt 200
       callback new Error 'unexpected status code'
-    else if body.indexOf('<title>Object moved</title>') >= 0
+    else if body.indexOf('<title>Object moved</title>') >= 0 and body.indexOf('Additional Sort:')
```
Comment (davidchambers) [inline src/gatherer.coffee]: It looks as though you're missing a `< 0` or a `>= 0`.

**Comment by CBXZero** [comment]
That "and" looked weird to me and the build test failed, if this build doesn't work I'll fix it when I can get to my laptop

**Comment by davidchambers** [comment]
Tests are failing. ;)

**Comment by CBXZero** [comment]
Looking at it on my dev machine now, I'll have it fixed here in a sec :)

**Comment by CBXZero** [comment]
Got it that time. I don't know why I thought that "and" looked funny. Then again, I've been studying Statistics all night and that might be messing with me. Lmk if there are any other changes you'd like done. :)

File: src/gatherer.coffee [hunk 2]
```diff
@@ -32,7 +32,7 @@ gatherer.request = (args...) ->
       callback err
     else if res.statusCode isnt 200
       callback new Error 'unexpected status code'
-    else if body.indexOf('<title>Object moved</title>') >= 0
+    else if body.indexOf('<title>Object moved</title>') >= 0 and body.indexOf('Additional Sort:') < 0
```
Comment (davidchambers) [inline src/gatherer.coffee]: If I remove this and run the tests, the test you added still passes. Apparently the new test does not actually test this.

**Comment by CBXZero** [comment]
I did some looking into this, and 2 things stood out to me.

1. I modified the javascript library by hand that my bot was using, and told it to just spit out the html of the file that it reads, and I noticed it hits the else block that is highlighted above multiple times (the file saves 3 times to be exact), and was wondering if you had any clues to that.

2. The and is incorrect, it should be an || body.indexOf(string) >= 0 . That being said, if I actually switch it to that, I get 27 tests that break, mostly involving an Uncaught AssertionError: {} === null, which makes me think the double sided cards are now getting caught on that else block.

I was wondering if there were any ideas that you had that might be causing the issues I'm running into. I'll investigate this issue more tomorrow.

**Comment by davidchambers** [comment]
I suggest looking at the differences between the responses to the following requests:

```
$ curl --silent http://gatherer.wizards.com/Pages/Card/Details.aspx?name=fizzbuzzldspla
```

```
$ curl --silent http://gatherer.wizards.com/Pages/Card/Details.aspx?name=test
```

**Comment by CBXZero** [comment]
I'm still looking into this, just haven't been able to come back to it due to an interview and hw. I'll work on it more this next week.

**Comment by davidchambers** [comment]
Any luck so far, @CBXZero?

**Comment by CBXZero** [comment]
Taking a look at it today, was swamped with schoolwork for a while. I should have something by the end of this weekend.

**Comment by CBXZero** [comment]
Spent some time on this during the weekend, the results that I get from curling the URL and the body that I console log when testing are different and that's currently what I'm hung up on.

**Comment by davidchambers** [comment]
If you provide a reproducible example of the difference in behaviour between `curl` and `node` I will help you to investigate. :)

**Comment by CBXZero** [comment]
I console.logged the body on the line right before the callback around line 38, pasted it into a test editor, and compared it to the fixture generated and they were radically different. I think there is some recursive call that is throwing me off.

**Comment by CBXZero** [comment]
Here are the bodies that get logged in order of when I console.log them (this is from a single tutor.card() call searching via name, in this case "test" was used):

https://gist.github.com/CBXZero/b9136fd563b7028e2165

https://gist.github.com/CBXZero/b3322224639489d4ce0b

https://gist.github.com/CBXZero/2e8ec5a0e98e844ac4ec

**Comment by davidchambers** [comment]
Superseded by #97

**Pull request closed without merging**